### ISSUE TITLE
Determine payload constraints and # of toastSats that fly

### ISSUE THREAD

**Issue by SeanTedesco** [queens-satellite-team/Space-School#8]

**Comment by spacemansears** [comment]
Here's an answer to the **mass** constraint based on some past experience...

_TL;DR: Each team has approximately 94 g allocated for their hardware._

Say we have an upper limit for the mass, `m_{max}`. We can determine the mass available for toastSats as `m_{max} = m_{toastSats} + m_{gondola}`

From my previous experience, we can calculate an estimate for `m_{gondola}`. That entry would be a sum of the following [source: August 20, 2019 flight]:

| Items | Mass (g) |
| --- | --- |
| Parachute | 253 |
| Tracker ([Tracksoar](https://www.tracksoar.com/)) and batteries | 64 |
| Tracker antenna supports | 17 |
| Payload box | 60 |
| Rope and flight train connections | 40 |
| Tape, zipties, elastic bands, other misc hardware | 30 |

This gives us a total of 464 g.

Next, what is `m_{max}`? This requires a little more understanding than a basic calculation to determine, as it is a balance between commercial availability, desired maximum altitude, desired ascent rate, and regulatory limitations on high altitude balloons. In fact, starting from scratch this would require an iterative process where we guess `m_{toastSats}`, but I can jump ahead to a reasonable solution.

For our planning, an 800 g balloon is a wise choice. This size is readily available and has a good capacity-to-weight ratio for a payload in our mass range. We will be within the regulatory limits for a balloon flight and ensure the flight is neither too long or too short to be fun.

Okay, but how does knowing the balloon size give us `m_{max}` you're wondering? The larger the balloon (measured in grams as it is the mass of the balloon), the more helium we can inflate it with. This increases the displacement of air and causes our upward force (buoyancy) to increase as well. We could calculate this by first principles for fun, but suitable calculators exist on the internet as well, such as this one: [habhub balloon calculator](http://habhub.org/calc/). 

How to use:
1. Pick a balloon size (800 g).
2. Erase "target burst altitude" and fill in "target ascent rate" as 5 m/s
   - this will give us a quick ascent and shorten flight duration (don't go below 5 m/s ideally)
3. Play with "payload mass" and see how the Helium volume requirements change.
4. Find the closest payload mass that gets the Helium requirement at or below 115 cubic feet [602.42](https://laws-lois.justice.gc.ca/eng/regulations/SOR-96-433/)

Assuming we use a "Kaymont - 800" g balloon, we predict 1317 g of payload mass @ 5 m/s ascent rate with 115 cubic feet of Helium. To be practical, we should give ourselves a little wiggle room, so knock off about 10% and say our `m_{max}` is 1200 g.

Almost there! Rearrange our mass sum at the top to solve for `m_{toastSats}` and we get `1200 - 464 = 736` g. We have four groups, so assume four toastSats so each team will have `736 / 4 = 184` g.

The mass budget for a toastSat will be something like: `m_{toastSat} = m_{arduino} + m_{breadboard} + m_{sensors} + m_{wires} + m_{power-system}`. We are providing the power system, which is an 18650 Lithium Ion battery, some sort of voltage regulator, and wires.

| Power system items | Mass (g) |
| --- | --- |
| 18650 2200 mAh battery | 64 |
| Voltage regulator | 6 |
| USB cable, A to micro B | 20 |

This will cost us 90 g, which is roughly half of the allocated toastSat mass. The teams therefore have **94 g** for their hardware.

**Comment by spacemansears** [comment]
**Power** constraints have been determined as well. 

A 2200 mAh 3.7 V Lithium Ion battery will be provided to each team for their payloads. This provides `2200 * 3.7 = 8140` mWh of energy. Knowing that the flight could be 4-5 hours, we can estimate the maximum average power consumption for a flight by `energy / time = average-power`, so `8410 / 5 = 1628` mW or 1.6 W.

Teams should take measurements of their payload to confirm they are drawing **less than 1.6 W**. 

Note that the voltage conversion is not 100% efficient, so 1.6 W is overly optimistic as there will be power lost in that process. Teams would ideally run their payload uninterrupted from full charge to dead and time how long it lasts (based on the microSD card recording perhaps).

**Comment by spacemansears** [comment]
In my last comment I said 2200 mAh, but Sean corrected me today and said it was 1200 mAh. That's a big drop, so we need teams to be closer to 0.8 W average power. Can re-do the analysis to confirm, but they should be building their power budgets themselves.

**Comment by spacemansears** [comment]
To do:
Volume constraint
